core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -552,6 +552,12 @@ public void testToQuery() throws IOException {
         }
     }
 
+    private QueryBuilder<?> rewriteQuery(QB queryBuilder, QueryRewriteContext rewriteContext) throws IOException {
+        QueryBuilder<?> rewritten = QueryBuilder.rewriteQuery(queryBuilder, rewriteContext);
+        assertSerialization(rewritten);
```

Comment:
if you rewrite into a state that is not serialized you are fucked and will never detect it